src/frontend/src/store/search/util/url-state-parser.ts

```diff
@@ -78,17 +79,76 @@ export default class UrlStateParser extends BaseUrlStateParser<HistoryModule.His
+	 */
+	@memoize
+	get spanFilters(): Record<string, FilterValue> {
+		const result: Record<string, FilterValue> = {};
+		Object.entries(this.withinClauses)
+			.forEach(([elName, attrs]) => {
+				Object.entries(attrs)
+					.forEach(([attrName, attrValue]) => {
+					const id = spanFilterId(elName, attrName);
+					const filter = FilterModule.getState().filters[id];
+					if (filter?.isSpanFilter) {
+						let values: string[];
+						if (typeof attrValue === 'string') {
+							if (filter.componentName === 'filter-select') {
+								// select, decode options
+								values = attrValue.split('|').map(v => unescapeRegex(v, { escapeWildcards: false }));
+							} else {
+								// text
+								values = [ unescapeRegex(attrValue, { escapeWildcards: false }) ];
+							}
+						} else if (attrValue.low || attrValue.high) {
+							values = [attrValue.low || '', attrValue.high || ''];
+						} else {
+							values = [ attrValue ];
+						}
+						result[id] = { id, values };
+					}
+				});
+		});
+		return result;
+	}
+
+	/** Within clauses that don't fit into a widget, so must remain part of the Expert query */
+	@memoize
+	get withinClausesWithoutSpanFilters(): Record<string, Record<string, any>> {
+		// Only keep within clauses that are not span filters
+		return Object.fromEntries(Object.entries(this.withinClauses)
+			.map(([elName, attrs]: [string, Record<string, any>]) => {
+				if (Object.keys(attrs).length === 0) {
+					// No attributes, so this might be the within widget selection.
+					return [elName, { '_MAYBE_WITHIN_': true } as Record<string, any>];
+				} else {
+					const filters = FilterModule.getState().filters;
+					const filteredAttrs = Object.fromEntries(Object.entries(attrs)
+						.filter(e => !filters[spanFilterId(elName, e[0])]?.isSpanFilter));
+					if (Object.keys(attrs).length > 0 && Object.keys(filteredAttrs).length === 0) {
+						// All attributes were placed in span filters, so we probably don't want this
+						// to be the within widget selection.
+						return [elName, { } as Record<string, any>];
+					} else {
+						return [elName, filteredAttrs as Record<string, any>];
+					}
+				}
+			})
+			.filter(([elName, attrs]: [string, Record<string, any>]) => Object.keys(attrs).length > 0)
+			.map(([elName, attrs]: [string, Record<string, any>]) => [elName, attrs['_MAYBE_WITHIN_'] ? {} : attrs])) as Record<string, Record<string, any>>;
+	}
+
 	@memoize
 	private get filters(): FilterModule.ModuleRootState {
 		const luceneString = this.getString('filter', null, v=>v?v:null);
-		if (luceneString == null) {
+		const spanFilters = this.spanFilters;
+		if (luceneString == null && Object.keys(spanFilters).length === 0) {
 			return {};
 		}
 
 		try {
-			const luceneQueryAST = LuceneQueryParser.parse(luceneString);
-			const parsedQuery: Record<string, FilterValue> = mapReduce(parseLucene(luceneString), 'id');
-
+			// FIXME: code below claims to be important but is never used!?
```

Comment:
Hmm, ik denk dat we over de allFilters moeten loopen in plaats van de Object.values(FilterModule.getState().filters) (regel 166). 
Zouden even in de git history moeten graven wat er hier aan de hand is.
Ik weet dat er ooit een probleem was met o.a. date-range filters (met een from en to veld op 2 verschillende metadata velden), waar de waardes 2x gedecode werd, 1 keer in de combinatiefilter en 1 keer in de losse filters op de metadata velden. Als je dan refreshte met zo'n setup kreeg je iedere keer 2 keer zo veel filter waardes.

Reply:
Okee, ik laat het in deze branch even met rust, als het aangepast moet worden, kunnen we dat in een aparte commit op dev doen.